.gitignore

```diff
@@ -0,0 +1,31 @@
+HELP.md
+target/
+!.mvn/wrapper/maven-wrapper.jar
```

Comment:
there is some stuff in this PR about maven wrapper (which we don't use in other projects), but I don't mind keeping it here if you are going to use it. The only benefit of keeping it is that it doesn't force the user to install maven.

Reply:
This .gitignore file was also generated by start.spring.io but wouldn't the user need maven to build this project anyway to resolve dependencies in pom.xml?

Reply:
the thing is, users have 2 options:
1. install maven (through apt-get for instance). then the maven wrapper stuff is not used.
2. use mvnw. this doesn't require the user to install maven. It checks if it's installed, and if not, mvnw downloads maven and uses it.